packages/application/src/connectionlost.ts

```diff
@@ -22,6 +25,45 @@ export const ConnectionLost: IConnectionLost = async function (
       'JupyterLab will continue trying to reconnect.\n' +
       'Check your network connection or Jupyter server configuration.\n'
   );
+  const buttons = [Dialog.okButton({ label: trans.__('Dismiss') })];
+
+  if (!displayConnectionLost) {
+    return;
+  }
+
+  const key = 'server-connection-error';
+  const promise = connectionLostCachePromise[key];
+
+  if (promise) {
+    return promise;
+  } else {
+    try {
+      const errorDialogPromise: any = showDialog({
+        title: title,
+        body: networkMsg,
+        checkbox: {
+          label: trans.__('Do not show this message again.'),
```

Comment:
I would be afraid to check this box thinking that I will never see the error again and that I won't find a way to re-enable it.

```suggestion
          label: trans.__('Do not show this message again in this session.'),
```